commands/add/index.js

```diff
@@ -130,6 +130,10 @@ class AddCommand extends Command {
       const deps = this.getPackageDeps(pkg);
       const range = getRangeToReference(this.spec, deps, pkg.location, this.savePrefix);
 
+      if (range === "^undefined") {
+        throw new Error("When adding a dependency, it must have a version.");
```

Comment:
This should be an instance of `ValidationError` (already imported in this file), otherwise it will dump a log and echo a stack, which really isn't what we want.